### PR TITLE
Feature/mesoscale plotting g2o

### PR DESCRIPTION
**Pull request by MarcelCaron-NOAA**
## Pull Request Testing ##

- [x] Describe testing already performed for this Pull Request:</br>

The mesoscale/grid2obs and /headline plots jobs with restart capability have been running cleanly for several weeks, and each run results in a tarball that contains the necessary graphics for EVS/mesoscale/grid2obs or /headline.  Restart capability has also been added to mesoscale/precip and /snowfall, and those jobs have been run successfully.  To test the restart capability, each job was run twice using the same VDATE.  The walltimes of the second runs were substantially shorter than they were for the first runs.

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

To test for both successful completion and restart capability, I recommend running the following jobs twice:
```
{HOMEevs}/dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_grid2obs_plots.sh
{HOMEevs}/dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_headline_plots.sh
{HOMEevs}/dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_precip_plots.sh
{HOMEevs}/dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_snowfall_plots.sh
```
...replacing {HOMEevs} with the location of the EVS testing directory.  No changes need to be made between the first and second runs. 

Before running, in the driver scripts:

1. Change COMIN to the following (i.e., change "${USER}" to "emc.vpppg"):
`export COMIN=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/$NET/$evs_ver/stats/$COMPONENT`
2. Predefine the VDATE in the driver script. Any VDATE should be fine.

The first runs should take between 4-9 hours to complete for the grid2obs, precip, and snowfall jobs and 10 minutes for the headline job.  Jobs will be checked for successful completion, and second jobs will be checked for reduced walltimes.

- [x] Has the code been checked to ensure that no errors occur during the execution? **[Yes]**

- [x] Do these updates/additions include sufficient testing updates? **[Yes]**

- [x] Please complete this pull request review by **[09/09/2023]**.</br>

## Pull Request Checklist ##

- [ ] Review the source issue metadata (required labels, projects, and milestone).
- [ ] Complete the PR description above.
- [ ] Ensure the PR title matches the feature branch name.
- [ ] Check the following:
- [ ]  Instructions provided on how to run
- [ ]  Developer's name is replaced by ${user} where necessary throughout the code
- [ ]  Check that the ecf file has all the proper definitions of variables
- [ ]  Check that the jobs file has all the proper settings of COMIN and COMOUT and other input variables
- [ ]  Check to see that the output directory structure is followed
- [ ]  Be sure that you are not using MET utilities outside the METplus wrapper structure

- [ ] After submitting the PR, select **Development** issue with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue.

### CODE REVIEW

File: dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_precip_plots.sh [hunk 1]
```diff
@@ -45,21 +45,20 @@ source /usr/share/lmod/lmod/init/sh
...
 export PYTHONPATH=$HOMEevs/ush/$COMPONENT:$PYTHONPATH
 
 # Developer Settings
-export DATA=/lfs/h2/emc/ptmp/$USER/evs_test/$envir/tmp
+export DATAROOT=/lfs/h2/emc/ptmp/$USER/evs_test/$envir/tmp
 export COMINccpa=/lfs/h2/emc/ptmp/${USER}/$NET/$evs_ver/prep/$COMPONENT/$RUN
 export COMINmrms=/lfs/h2/emc/ptmp/${USER}/$NET/$evs_ver/prep/$COMPONENT/$RUN
 export COMINspcotlk=/lfs/h2/emc/ptmp/${USER}/$NET/$evs_ver/prep/$COMPONENT/$RUN
-export COMIN=/lfs/h2/emc/vpppg/noscrub/${USER}/$NET/$evs_ver/stats/
+export COMIN=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/$NET/$evs_ver/stats/
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_precip_plots.sh]: We'll want this changed back to $USER for the repository.

**Comment by ShelleyMelchior-NOAA** [comment]
@MarcelCaron-NOAA , I have cloned your fork and checked out branch `feature/mesoscale_plotting_g2o`.

I set VDATE=20230831 and ran all 4 plots drivers.

headline completed quickly, as you indicated, with a runtime of ~5.5 minutes.
COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/mesoscale/atmos.20230831`
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_headline_plots.o95847969`
The re-run of headline, to test restart, completed very quickly, in about 30 seconds.
logfile: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_headline_plots.o95851756`

The other three plots processes are still running for the first submissions. I will update when the other 3 processes conclude.

**Comment by ShelleyMelchior-NOAA** [comment]
The grid2obs, precip, and snowfall original runs for VDATE=20230831 have completed.
logfiles: `/lfs/h2/emc/ptmp/shelley.melchior/EVS_out/jevs_mesoscale_*_plots.o95847*`

COMOUT=`/lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/mesoscale/atmos.20230831` (grid2obs, headline)
```
$ ll /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/mesoscale/atmos.20230831
total 14706848
-rw-r--r-- 1 shelley.melchior vpppg 15043123200 Sep  4 00:55 evs.plots.mesoscale.atmos.grid2obs.v20230831.tar
-rw-r--r-- 1 shelley.melchior vpppg    16640000 Sep  3 19:42 evs.plots.mesoscale.atmos.headline.v20230831.tar
```
COMOUT=`/lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/mesoscale/atmos.20230831` (precip, snowfall)
```
$ ll /lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/mesoscale/atmos.20230831
total 1979428
-rw-r--r-- 1 shelley.melchior emc 1685309440 Sep  4 00:26 evs.plots.mesoscale.atmos.precip.v20230831.tar
-rw-r--r-- 1 shelley.melchior emc  341606400 Sep  4 01:56 evs.plots.mesoscale.atmos.snowfall.v20230831.tar
```

I re-ran for the same VDATE and noted the run time differences b/t the original and the re-run.
precip orig took 6:09; precip re-run took 0:22
snowfall orig took 7:39; snowfall re-run took 0:37
grid2obs orig took 6:38; grid2obs re-run took 0:29

Re-listing on the COMOUT locations you can see the updated timestamp but same file sizes.
```
$ ll /lfs/h2/emc/vpppg/noscrub/shelley.melchior/evs/v1.0/plots/mesoscale/atmos.20230831
total 14706860
-rw-r--r-- 1 shelley.melchior vpppg 15043123200 Sep  4 12:49 evs.plots.mesoscale.atmos.grid2obs.v20230831.tar
-rw-r--r-- 1 shelley.melchior vpppg    16640000 Sep  3 19:42 evs.plots.mesoscale.atmos.headline.v20230831.tar
```
and
```$ ll /lfs/h2/emc/ptmp/shelley.melchior/evs/v1.0/plots/mesoscale/atmos.20230831
total 1979436
-rw-r--r-- 1 shelley.melchior emc 1685309440 Sep  4 12:42 evs.plots.mesoscale.atmos.precip.v20230831.tar
-rw-r--r-- 1 shelley.melchior emc  341606400 Sep  4 12:57 evs.plots.mesoscale.atmos.snowfall.v20230831.tar
```

**Comment by MarcelCaron-NOAA** [comment]
Hi @ShelleyMelchior-NOAA, thanks for running these and recording their results this weekend.  These ran as expected--completing cleanly, outputting tar files similar to those I ran in my tests, with substantially reduced walltimes for the second runs.  Looks good to me!

File: dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_precip_plots.sh (hunk 1 above)
Comment (MarcelCaron-NOAA) [reply dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_precip_plots.sh]: Changing this now

**Comment by ShelleyMelchior-NOAA** [comment]
@MarcelCaron-NOAA , fantastic! I will look for the small code change and go through the formal code review.

File: dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_grid2obs_plots.sh [hunk 2]
```diff
@@ -1 +1,66 @@
...
+export MET_PLUS_PATH="/apps/ops/para/libs/intel/${intel_ver}/metplus/${metplus_ver}"
+export MET_PATH="/apps/ops/para/libs/intel/${intel_ver}/met/${met_ver}"
+export MET_CONFIG="${MET_PLUS_PATH}/parm/met_config"
+export PYTHONPATH=$HOMEevs/ush/$COMPONENT:$PYTHONPATH
+
+# Developer Settings
+export DATAROOT=/lfs/h2/emc/stmp/$USER/evs_test/$envir/tmp
+export COMINccpa=/lfs/h2/emc/ptmp/${USER}/$NET/$evs_ver/prep/$COMPONENT/$RUN
+export COMINmrms=/lfs/h2/emc/ptmp/${USER}/$NET/$evs_ver/prep/$COMPONENT/$RUN
+export COMINspcotlk=/lfs/h2/emc/ptmp/${USER}/$NET/$evs_ver/prep/$COMPONENT/$RUN
```
Comment (ShelleyMelchior-NOAA) [inline dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_grid2obs_plots.sh]: Are these 3 COMIN* variables needed? In my test runs I only changed COMIN to point to emc.vpppg's stats. I left these 3 COMIN* variables unchanged and everything ran fine. I am wondering if they are an artifact of an inherited cam driver script.

Comment (MarcelCaron-NOAA) [reply dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_grid2obs_plots.sh]: The three COMIN variables are not needed and they are an artifact of an earlier driver script.  I plan to remove these in a clean-up PR for meso/cam so I can test what can be removed.  However these can definitely be removed in this PR instead if that's preferred.

Comment (ShelleyMelchior-NOAA) [reply dev/drivers/scripts/mesoscale/plots/jevs_mesoscale_grid2obs_plots.sh]: Thanks @MarcelCaron-NOAA ! Yes, let's remove for the mesoscale PR.

**Review by ShelleyMelchior-NOAA — APPROVED** [review]
All changes were tested and perform as expected. This PR is approved for merge.

**Pull request merged**